loopd/swapclient_server.go

```diff
@@ -920,3 +927,24 @@ func validateLoopInRequest(htlcConfTarget int32, external bool) (int32, error) {
 
 	return validateConfTarget(htlcConfTarget, loop.DefaultHtlcConfTarget)
 }
+
+// validateLoopOutRequest validates the confirmation target, destination
+// address and label of the loop out request.
+func validateLoopOutRequest(chainParams *chaincfg.Params, confTarget int32,
+	sweepAddr btcutil.Address, label string) (int32, error) {
+
+	// Check that the provided destination address has the correct format
+	// for the active network.
+	if !sweepAddr.IsForNet(chainParams) {
+		log.Errorf("destination address format does not match "+
+			"the active network: %s", chainParams.Name)
+		return 0, errIncorrectChain
```

Comment:
Similarly to conf target I'd suggest returning a detailed error instead here too.

Reply:
Detailed error makes this a little harder to test, perhaps add a `newChainError` func which returns the detailed error and can be used to match exactly in the test? Otherwise could define different types and use `errors.As`, but that's probably overkill.